lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecomposition.cpp

```diff
@@ -224,34 +224,27 @@ static std::vector<TransposeDims> generateConvKernelTransposeIndices(
   return generateTransposeIndices(kernelLayout, ttnnConvolutionKernelLayout);
 }
 
-struct ConvolutionToConv2dPattern
+struct ConvolutionDecompositionPattern
     : public OpConversionPattern<ttir::ConvolutionOp> {
 public:
   using OpConversionPattern<ttir::ConvolutionOp>::OpConversionPattern;
 
-  constexpr static uint32_t numSpatialDims = 2;
-  constexpr static uint32_t SPATIAL_DIM_HEIGHT = 0;
-  constexpr static uint32_t SPATIAL_DIM_WIDTH = 1;
-
-  // NHWC
-  static inline const std::vector<int64_t> conv2dLayout = {
-      ConvolutionDimension::BATCH,
-      SPATIAL_DIM_HEIGHT,
-      SPATIAL_DIM_WIDTH,
-      ConvolutionDimension::FEATURE,
-  };
-  // OIHW
-  static inline const std::vector<int64_t> conv2dKernelLayout = {
-      ConvolutionKernelDimension::OUTPUT_FEATURES,
-      ConvolutionKernelDimension::INPUT_FEATURES,
-      SPATIAL_DIM_HEIGHT,
-      SPATIAL_DIM_WIDTH,
-  };
+  LogicalResult
+  matchAndRewrite(ttir::ConvolutionOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override = 0;
 
-  LogicalResult isConv2d(ttir::ConvolutionOp op) const {
+protected:
+  LogicalResult isNDimensional(ttir::ConvolutionOp op,
```

Comment:
What's the value of using `LogicalResult` instead of `bool` for these utility functions?

Reply:
You are correct, there is no added value. Changed to `bool`.